lib/cog/repository/groups.ex

```diff
@@ -0,0 +1,215 @@
+defmodule Cog.Repository.Groups do
+  @moduledoc """
+  Behavioral API for interacting with user groups. Prefer these
+  functions over direct manipulation with 'Cog.Repo'.
+  """
+
+  alias Cog.Repo
+  alias Cog.Models.Group
+  alias Cog.Models.User
+  alias Cog.Models.Role
+  import Ecto.Query, only: [from: 1, from: 2]
+
+
+  @doc """
+  Creates a new user group given a map of attributes
+  """
+  @spec new(Map.t) :: {:ok, %Group{}} | {:error, Ecto.Changeset.t}
+  def new(attrs) do
+    %Group{}
+    |> Group.changeset(attrs)
+    |> Repo.insert
+  end
+
+  @doc """
+  Retrieves all user groups
+  """
+  @spec all :: [%Group{}]
+  def all,
+    do: Repo.all(Group) |> Repo.preload([:direct_user_members, :direct_group_members, :roles])
```

Comment:
I don't think you need `direct_group_members` anymore, since groups can't be members of groups.

If you can remove this and the tests still pass, go ahead and take it out. I know there are some places in our test setup code that still might take advantage of this. If that's the case, can you make a note to that effect here, so we can take out the preload once we tighten the tests up / remove all the outdated authz infrastructure?